### PR TITLE
Create BareRootProductGetterBase for FireworksWeb

### LINKED ISSUE #987: Create BareRootProductGetterBase for FireworksWeb

### PR DESCRIPTION
**Pull request by makortel**
#### PR description:

@alja had a question on if `BareRootProductGetter.h` could be made a public header in `FWCore/FWLite` package, because FireworksWeb needs to use a different mechanism to provide the current file than what `BareRootProductGetter` implements. I created a customization point for FireworksWeb by moving nearly all functionality of `BareRootProductGetter` into a new `BareRootProductGetterBase`, except the logic to get the current file in `BareRootProductGetterBase::getIt()`. I added a new pure virtual function `currentFile()`, that is overridden and implemented in `BareRootProductGetter`. FireworksWeb can now provide a different implementation by inheriting from `BareRootProductGetterBase` and implementing only the `currentFile()`.

Resolves https://github.com/cms-sw/framework-team/issues/987

#### PR validation:

Code compiles, unit tests pass.

### CODE REVIEW

**Comment by makortel** [comment]
@cmsbuild, please test

File: FWCore/FWLite/src/BareRootProductGetterBase.cc [hunk 1]
```diff
@@ -0,0 +1,249 @@
+// -*- C++ -*-
+//
+// Package:     FWLite
+// Class  :     BareRootProductGetterBase
+//
+// Implementation:
+//     <Notes on implementation>
+//
+// Original Author:  Chris Jones
+//         Created:  Tue May 23 11:03:31 EDT 2006
```
Comment (Dr15Jones) [inline FWCore/FWLite/src/BareRootProductGetterBase.cc:10]: Given this won't be consistent with the git history. How about you extend the comment say that you copied the original.

File: FWCore/FWLite/interface/BareRootProductGetterBase.h [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+/**\class BareRootProductGetterBase BareRootProductGetterBase.h FWCore/FWLite/interface/BareRootProductGetterBase.h
+
+ Description: <one line class summary>
+
+ Usage:
+    <usage>
+
+*/
+//
+// Original Author:  Chris Jones
```
Comment (Dr15Jones) [inline FWCore/FWLite/interface/BareRootProductGetterBase.h:17]: How about updating this comment to say the code was moved so the git history doesn't conflict with this comment?

**Comment by alja** [comment]
+1

**Comment by alja** [comment]
Thank you Matti for  the implementation!

This change is needed for the event filtering in FireworksWeb. The legacy Fireworks has the event filtering in the main thread, but FireworksWeb has event filtering in  the separate thread [1].

[1]https://github.com/alja/FireworksWeb/blob/main/Core/src/CmsShowNavigator.cc#L406-L409

File: FWCore/FWLite/src/BareRootProductGetterBase.cc (hunk 1 above)
Comment (makortel) [reply FWCore/FWLite/src/BareRootProductGetterBase.cc:14]: How about now (in both `.cc` and `.h`)?

**Comment by makortel** [comment]
Comparison differences are related to https://github.com/cms-sw/cmssw/issues/39803

**Comment by makortel** [comment]
@cmsbuild, please test

**Comment by Dr15Jones** [comment]
looks good

**Comment by makortel** [comment]
+core

**Comment by mandrenguyen** [comment]
+1

**Pull request merged**